### PR TITLE
Export WriteStream and switch constructor args

### PR DESCRIPTION
**Pull request by rvagg**
Now just exports `WriteStream` to encourage users not to monkey patch (I think we need to get away from this practice), but I've replaced it with a `patch()` export that'll put `createWriteStream()` back on if you want it.

Also switched the arguments to put the optional `options` object second:

```
var ws = new WriteStream(db, options)
```

Fixed up tests, did some minor formatting, cleaned up dependencies.

Ready for a 1.0.0 unless I get some disagreement.

### CODE REVIEW

**Comment by ralphtheninja** [comment]
lgtm

**Comment by jcrugzz** [comment]
LGTM as well

**Comment by ralphtheninja** [comment]
I'm rebasing this onto latest master and adding some commits to it

**Comment by jcrugzz** [comment]
@ralphtheninja :+1:

**Comment by ralphtheninja** [comment]
Oh wow this has been here for a long time. Lets see if we can salvage some of this code.

File: test.js [hunk 1]
```diff
@@ -67,7 +68,7 @@ function setUp (t) {
     })
   }
 
-  this.verify = function (ws, db, done, data) {
+  this.verify = delayed(function (ws, db, done, data) {
```
Comment (ralphtheninja) [inline test.js]: @vweevers I'm wondering if this was to "fix" the test race conditions.

**Comment by ralphtheninja** [comment]
Read through this PR again and it's really hard to rebase, so I'll pick the raisins out of the cake and re-implement. Changing the API is one thing, see https://github.com/Level/level-ws/issues/55

**Comment by ralphtheninja** [comment]
https://github.com/Level/level-ws/issues/55

**Pull request closed without merging**